datadog_lambda/tracing.py

```diff
@@ -106,12 +106,15 @@ def create_dd_dummy_metadata_subsegment(
     tags into its metadata field, so the X-Ray trace can be converted to a Datadog
     trace in the Datadog backend with the correct context.
     """
-    xray_recorder.begin_subsegment(XraySubsegment.NAME)
-    subsegment = xray_recorder.current_subsegment()
-    subsegment.put_metadata(
-        subsegment_metadata_key, subsegment_metadata_value, XraySubsegment.NAMESPACE
-    )
-    xray_recorder.end_subsegment()
+    try:
+        xray_recorder.begin_subsegment(XraySubsegment.NAME)
+        subsegment = xray_recorder.current_subsegment()
+        subsegment.put_metadata(
+            subsegment_metadata_key, subsegment_metadata_value, XraySubsegment.NAMESPACE
+        )
+        xray_recorder.end_subsegment()
+    except Exception as e:
+        logger.debug("failed to create dd dummy metadata subsegment with error %s", e)
```

Comment:
I think it's essential to log the entire stack trace https://github.com/DataDog/datadog-lambda-python/blob/1ef99459a0b4f50931471106c089783926158578/datadog_lambda/metric.py#L102